storage/innobase/include/srw_lock.h

```diff
@@ -223,12 +228,7 @@ class ssux_lock_impl
 
   bool u_lock_try()
   {
-    if (!writer.wr_lock_try())
-      return false;
-    IF_DBUG_ASSERT(uint32_t lk=,)
-    readers.fetch_add(1, std::memory_order_acquire);
-    DBUG_ASSERT(lk < WRITER - 1);
-    return true;
+    return writer.wr_lock_try();
   }
 
   bool wr_lock_try()
```

Comment:
`ssux_lock_impl::wr_lock_try()` had better be refactored so that we can avoid unnecessary calls to `writer.wr_unlock()` as part of blocking operations such as `dict_sys_t::lock()`.

Reply:
To check this, I renamed various definitions of `wr_lock_try()` and checked every caller in the chain. We can improve `psi_wr_lock()` so that if only the `lock.writer.wr_lock_try()` succeeds but some shared locks in `lock.readers` exist, we will retain the `lock.writer` latch and invoke `lock.u_wr_upgrade()` after registering the wait in `PERFORMANCE_SCHEMA`.

`dict_sys.lock(SRW_LOCK_CALL)` calls are rather rare, mostly occurring during database startup and DDL. The only calls during a DML-only workload should be related to loading tables into or evicting them from the `dict_sys` cache. Because `dict_sys.latch` can also be `SRWLOCK` or `rw_lock_t` and because the exclusive lock requests are rare, it it probably does not make sense to optimize that case.

Reply:
I don’t think that it makes sense to improve `dict_sys_t::lock()`. Only `psi_wr_lock()` can be improved: 509d5e761a4db8b598616b4165f992f08c319c69. All other callers of any `wr_lock_try()` would back off more and not proceed to directly wait for the exclusive lock.